src/analysis/calculus/uniform_limits_deriv.lean

```diff
@@ -0,0 +1,567 @@
+That is, we want to prove something like:
+
+```
+∀ ε > 0, ∃ δ > 0, ∀ y ∈ B_δ(x), |y - x|⁻¹ * |(g y - g x) - g' x (y - x)| < ε.
+```
+
+To do so, we will need to introduce a pair of quantifers
+
+```lean
+∀ ε > 0, ∃ N, ∀ n ≥ N, ∃ δ > 0, ∀ y ∈ B_δ(x), |y - x|⁻¹ * |(g y - g x) - g' x (y - x)| < ε.
+```
+
+So how do we write this in terms of filters? Well, the initial definition of the derivative is
+
+```lean
+tendsto (|y - x|⁻¹ * |(g y - g x) - g' x (y - x)|) (𝓝 x) (𝓝 0)
+```
+
+There are two ways we might introduce `n`. We could do:
+
+```lean
+∀ᶠ (n : ℕ) in at_top, tendsto (|y - x|⁻¹ * |(g y - g x) - g' x (y - x)|) (𝓝 x) (𝓝 0)
+```
+
+but this is equivalent to the quantifier order `∃ N, ∀ n ≥ N, ∀ ε > 0, ∃ δ > 0, ∀ y ∈ B_δ(x)`,
+which _implies_ our desired `∀ ∃ ∀ ∃ ∀` but is _not_ equivalent to it. On the other hand, we might
+try
+
+```lean
+tendsto (|y - x|⁻¹ * |(g y - g x) - g' x (y - x)|) (at_top ×ᶠ 𝓝 x) (𝓝 0)
+```
+
+but this is equivalent to the quantifer order `∀ ε > 0, ∃ N, ∃ δ > 0, ∀ n ≥ N, ∀ y ∈ B_δ(x)`, which
+again _implies_ our desired `∀ ∃ ∀ ∃ ∀` but is not equivalent to it.
+
+So to get the quantifier order we want, we need to introduce a new filter construction, which we
+call a "curried filter"
+
+```lean
+tendsto (|y - x|⁻¹ * |(g y - g x) - g' x (y - x)|) (at_top.curry (𝓝 x)) (𝓝 0)
+```
+
+Then the above implications are `filter.tendsto.curry` and
+`filter.tendsto.mono_left filter.curry_le_prod`. We will use both of these deductions as part of
+our proof.
+
+We note that if you loosen the assumptions of the main theorem then the proof becomes quite a bit
+easier. In particular, if you assume there is a common neighborhood `s` where all of the three
+assumptions of `has_fderiv_at_of_tendsto_uniformly_on_filter` hold and that the `f'` are
+continuous, then you can avoid the mean value theorem and much of the work around curried fitlers.
+
+## Tags
+
+uniform convergence, limits of derivatives
+-/
+
+open filter
+open_locale uniformity filter topological_space
+
+section limits_of_derivatives
+
+variables {ι : Type*} {l : filter ι} [ne_bot l]
+  {E : Type*} [normed_add_comm_group E] [normed_space ℝ E]
+  {𝕜 : Type*} [is_R_or_C 𝕜] [normed_space 𝕜 E]
+  {G : Type*} [normed_add_comm_group G] [normed_space 𝕜 G]
+  {f : ι → E → G} {g : E → G} {f' : ι → (E → (E →L[𝕜] G))} {g' : E → (E →L[𝕜] G)}
+  {x : E}
+
+/-- If a sequence of functions real or complex functions are eventually differentiable on a
+neighborhood of `x`, they converge pointwise _at_ `x`, and their derivatives
+converge uniformly in a neighborhood of `x`, then the functions form a uniform Cauchy sequence
+in a neighborhood of `x`. -/
+lemma uniform_cauchy_seq_on_filter_of_tendsto_uniformly_on_filter_fderiv
+  (hf : ∀ᶠ (n : ι × E) in (l ×ᶠ 𝓝 x), has_fderiv_at (f n.fst) (f' n.fst n.snd) n.snd)
+  (hfg : tendsto (λ n, f n x) l (𝓝 (g x)))
+  (hfg' : uniform_cauchy_seq_on_filter f' l (𝓝 x)) :
```

Comment:
I recommend reordering the arguments to make `hfg'` come before `hf` (and maybe `hfg'` should be called `hf'`?). The reason is that Lean has trouble figuring out what `f'` is from the expression `f' n.fst n.snd` (when it appears in the wild), but has no problem figuring it out from `uniform_cauchy_seq_on_filter f' l (𝓝 x)`. This holds for all(?) lemmas below as well.

Reply:
Done and for all the above